diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2268d852..5c0ee707 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -315,7 +315,9 @@ IF(CMAKE_HOST_UNIX AND NOT FORCE_UNSUPPORTED_COMPILER
         ELSE()
           SET(ALTERNATIVE_PATHS "/opt/rh/devtoolset-11")
         ENDIF()
-      ELSEIF(LINUX_RHEL8 OR LINUX_RHEL9)
+      ELSEIF(LINUX_RHEL8)
+        SET(ALTERNATIVE_PATHS "/opt/rh/gcc-toolset-11")
+      ELSEIF(LINUX_RHEL9)
         SET(ALTERNATIVE_PATHS "/opt/rh/gcc-toolset-12")
       ENDIF()
 
@@ -368,7 +370,7 @@ IF(CMAKE_HOST_UNIX AND NOT FORCE_UNSUPPORTED_COMPILER
         MESSAGE(STATUS "Using ${ALTERNATIVE_GCC}")
         MESSAGE(STATUS "Using ${ALTERNATIVE_GPP}")
       ELSE()
-        IF(LINUX_RHEL7)
+      IF(LINUX_RHEL7)
           IF(MY_HOST_MACHINE_NAME MATCHES "aarch64")
             SET(DEV_PACKAGES
               "devtoolset-10-gcc devtoolset-10-gcc-c++ devtoolset-10-binutils")
@@ -376,7 +378,12 @@ IF(CMAKE_HOST_UNIX AND NOT FORCE_UNSUPPORTED_COMPILER
             SET(DEV_PACKAGES
               "devtoolset-11-gcc devtoolset-11-gcc-c++ devtoolset-11-binutils")
           ENDIF()
-        ELSEIF(LINUX_RHEL8 OR LINUX_RHEL9)
+        ELSEIF(LINUX_RHEL8)
+          SET(DEV_PACKAGES
+            "gcc-toolset-11-gcc gcc-toolset-11-gcc-c++ gcc-toolset-11-binutils")
+          STRING_APPEND(DEV_PACKAGES " gcc-toolset-11-annobin-annocheck")
+          STRING_APPEND(DEV_PACKAGES " gcc-toolset-11-annobin-plugin-gcc")
+        ELSEIF(LINUX_RHEL9)
           SET(DEV_PACKAGES
             "gcc-toolset-12-gcc gcc-toolset-12-gcc-c++ gcc-toolset-12-binutils")
           STRING_APPEND(DEV_PACKAGES " gcc-toolset-12-annobin-annocheck")
